lib/src/configuration.dart

```diff
@@ -27,14 +26,32 @@ class Configuration {
   final Map<String, ConfiguredValue> _values;
 
   /// Creates an implicit configuration with the given [values].
-  Configuration.implicit(this._values);
+  Configuration.implicit(this._values) : _originalConfiguration = null;
+
+  /// Reference to the original configuration serving as an opaque ID.
+  final Configuration? _originalConfiguration;
+
+  /// Returns whether [this] and [that] [Configuration]s have the same
+  /// [_originalConfiguration].
+  ///
+  /// An implicit configuration will always return `false` because it was not
+  /// created through another configuration.
+  ///
+  /// [ExplicitConfiguration]s will and configurations created [throughForward]
+  /// will be considered to have the same original config if they were created
+  /// as a copy from the same base configuration.
+  bool sameOriginal(Configuration that) =>
```

Comment:
It seems like this should be reflexive—that is, a configuration should be said to have the same original configuration as itself. Maybe `.implicit()` and `.empty()` should set `_originalConfiguration` to `this` instead of null? That would allow you to remove some complexity with the `withValues()`/`copy()` method, too.

Reply:
Yup, I actually tried a few ways to achieve "this" earlier but a constructor initializer list can't access `this`, and if I declare it in the constructor body I must declare the originalConfiguration as `late final` which disallows `const` to be used in `Configuration.empty()`

Another alternative is renaming back to `opaqueId` and either assign `Object()` or `SomeNewClass()` value to it

Reply:
Good point. I'd use a private getter backed by a field here:

```dart
  /// The configuration from which this was modified with `@forward ... with`.
  Configuration get _originalConfiguration => __originalConfiguration ?? this;

  /// The backing value for [_originalConfiguration].
  ///
  /// This is null if [_originalConfiguration] refers to this configuration,
  /// since `this` can't be assigned to a final field.
  final Configuration? __originalConfiguration;
```